### ISSUE TITLE
What about ChronicleMap support ?

### ISSUE THREAD

**Issue by eix128** [bucket4j/bucket4j#267]
Hi , will you support ChronicleMap for jcache ?
its the fastest distrubuted map for java

**Comment by vladimir-bukhtoyarov** [comment]
Does ChronicleMap support JCache specification? I was unable to find something about JCache in its documentation.

**Comment by vladimir-bukhtoyarov** [comment]
I have investigated the documentation of ChronicleMap and made conclusion that there is no reasons to add integration for it, because of following limitations:
* ChronicleMap does not provide entry expiration feature, and in same time it does not provide secondary indexes(that can be used for indexing by TTL), so there is no way to effectively implement the process of cleaning obsolete buckets.
* For distributed use-case, ChronicleMap does not provide transaction isolation, two independent process on different hosts are allowed to mutate entries in same time independently from each other and this will lead to missing updates on conflict, because timestamp based conflict resolution is not suitable  for token-bucket.   Also it is impossible to configure custom conflict resolving handler for distributed replication, so it is even not impossible to achieve eventually consistence.

In conclusion I want to recommend to use:
* Bucket4j-Caffeine integration for single node use-case.
* And one of Bucket4j integration with distributed computational platforms like Coherence, Infinispan, Hazelcast or Ignite for case when you need for distributed buckets.
